src/archivist/services/index.js

```diff
@@ -4,261 +4,284 @@ import { pathToFileURL } from 'url';
+      contentSelectors: sourceDocumentDeclaration.select ?? termsDeclaration.select,
+      insignificantContentSelectors: sourceDocumentDeclaration.remove ?? termsDeclaration.remove,
+      filters: await getServiceFilters(serviceFilters, sourceDocumentDeclaration.filter ?? termsDeclaration.filter),
+    })));
+}
+
+export async function loadServiceFilters(serviceId) {
+  const serviceFiltersPath = path.join(declarationsPath, `${serviceId}${FILTERS_SUFFIX}${JS_EXT}`);
+
+  if (await fileExists(serviceFiltersPath)) {
+    return import(pathToFileURL(serviceFiltersPath));
+  }
 
-  return filterNames.map(filterName => serviceFilters[filterName]);
+  return {};
 }
 
-async function loadServiceDocument(service, termsType, termsTypeDeclaration) {
-  const { filter: filterNames, fetch: location, executeClientScripts, select: contentSelectors, remove: insignificantContentSelectors, combine } = termsTypeDeclaration;
-
-  const sourceDocuments = [];
-
-  const filters = await loadServiceFilters(service.id, filterNames);
-
-  if (!combine) {
-    sourceDocuments.push(new SourceDocument({ location, executeClientScripts, contentSelectors, insignificantContentSelectors, filters }));
-  } else {
-    for (const sourceDocument of combine) {
-      const {
-        filter: sourceDocumentFilterNames,
-        fetch: sourceDocumentLocation,
-        executeClientScripts: sourceDocumentExecuteClientScripts,
-        select: sourceDocumentContentSelectors,
-        remove: sourceDocumentInsignificantContentSelectors,
-      } = sourceDocument;
-
-      const sourceDocumentFilters = await loadServiceFilters(service.id, sourceDocumentFilterNames);
-
-      sourceDocuments.push(new SourceDocument({
-        location: sourceDocumentLocation || location,
-        executeClientScripts: (sourceDocumentExecuteClientScripts === undefined || sourceDocumentExecuteClientScripts === null ? executeClientScripts : sourceDocumentExecuteClientScripts),
-        contentSelectors: sourceDocumentContentSelectors || contentSelectors,
-        insignificantContentSelectors: sourceDocumentInsignificantContentSelectors || insignificantContentSelectors,
-        filters: sourceDocumentFilters || filters,
-      }));
-    }
+export function getServiceFilters(serviceFilters, filterNames) {
+  if (!filterNames) {
+    return undefined;
   }
 
-  service.addTerms(new Terms({ service, type: termsType, sourceDocuments }));
+  const filters = filterNames.reduce((filters, filterItem) => {
+    let filterFunction;
+
+    if (typeof filterItem === 'string') {
+      filterFunction = exposedFilters[filterItem] || serviceFilters[filterItem];
+    } else if (typeof filterItem === 'object' && filterItem !== null) {
+      const [ filterName, params ] = Object.entries(filterItem)[0];
+      const baseFunction = exposedFilters[filterName] || serviceFilters[filterName];
+
+      if (baseFunction) {
```

Comment:
So we silently fail if the filter name has no match? Shouldn't we throw a validation error if the user tries to apply a filter that does not exist? Most likely it is a typo and we will save them time; if not, it is likely a misunderstanding of the usage of filters, such as copy-pasting another declaration without having the appropriate filter file, and we should inform them as well.